src/library.js

```diff
@@ -2700,6 +2713,14 @@ LibraryManager.library = {
   emscripten_longjmp: function(env, value) {
     _longjmp(env, value);
   },
+#if WASM_BACKEND == 1
+  // We need this wrapper from llvm wasm backend, but this is generated only
+  // in asm.js compilation, so.
```

Comment:
I don't know why it is done that way, but in asm.js, the call sequence is
emscripten_longjmp__wrapper(env, value) -> emscripten_longjmp(env, value) -> _longjmp(env, value).

Coincidentally, wasm llvm pass needs this call sequence too, because in LLVM IR side we need emscripten_wrapper for two kinds of function signatures. The first one is 
emscripten_longjmp(jmp_buf, int), which is the same as the original longjmp. We need this because we have to replace calls to longjmp to this. But secondly we also need emscripten_longjmp(int, int), because we need to model this code in LLVM side:
https://github.com/kripken/emscripten-fastcomp/blob/incoming/lib/Target/JSBackend/CallHandlers.h#L229
Here we call emscripten_longjmp with its first argument int (the value loaded from __THREW__ global variable). In asm.js, this code is in JS side so it doesn't matter, but we need to do this in llvm, so we need to call a function with the signature (int, int).

That's the reason why we need two signatures of emscripten_longjmp in llvm pass. Currently, in the LLVM pass, I declare functions as
emscripten_longjmp__wrapper(jmp_buf, int)
emscripten_longjmp(int, int)
And emscrpten_longjmp__wrapper calls emscripten_longjmp, which calls _longjmp. emscripten_longjmp__wrapper's signature will be lowered to (int, int) eventually in the backend, but in the LLVM IR pass we need two separate signatures.